code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/ContentCardSchemaDataUtils.kt

```diff
@@ -134,41 +154,79 @@ internal object ContentCardSchemaDataUtils {
         }
     }
 
-    private fun createSmallImageTemplate(contentMap: Map<String, Any>, id: String): AepUITemplate? {
-        val title = DataReader.getTypedMap(Any::class.java, contentMap, MessagingConstants.ContentCard.UIKeys.TITLE)?.let {
-            createAepText(it)
-        }
-        if (title == null || title.content.isEmpty()) {
-            Log.error(
-                MessagingConstants.LOG_TAG,
-                SELF_TAG,
-                "Title is missing in the content card with id: $id. Skipping the content card."
-            )
-            return null
-        }
-        val body = DataReader.optTypedMap(Any::class.java, contentMap, MessagingConstants.ContentCard.UIKeys.BODY, null)?.let {
-            createAepText(it)
-        }
+    private fun createAepUITemplate(templateType: AepUITemplateType, contentMap: Map<String, Any>, id: String): AepUITemplate? {
```

Comment:
I recommend creating individual templates instead of a blanket method with clauses in it. I say that because I feel the logic is easier to follow that way without having to worry about mixing up the clauses and affecting other templates later on.